autotest/autotest.py

```diff
@@ -94,16 +110,43 @@ def ansi(code):
 
 keymaps = { "^C": "\x03" }
 
-while True:
+wait_for, stop_action, method = "", "", ""
+
+interactive = True
+if os.environ.get("WORKSHOP_TEST_FORCE_NONINTERACTIVE"):
+    interactive=False
+
+while i < len(actions):
     with open("nextstep","w") as f:
         f.write(str(i))
     slide, snippet, method, data = actions[i]
     data = data.strip()
+
+    # Look behind at the last slide to see if 'wait' or 'keys' was defined.
+    # If so, we need to wait for the specified output and/or terminate the command with the specified keys.
+    if method == "wait":
+        wait_for = data
+        print("Setting wait_for to: {}".format(data))
+        i += 1
+        continue
+    if method == "keys":
```

Comment:
Why do we need a special handling for `keys`? Normally we should be able to just flow naturally to the next snippet, no?